### DISCUSSION TITLE
Conditionally prevent deleting a Repeater entry

### DISCUSSION THREAD

**Discussion by eithed** [filamentphp/filament#9912] [Help]
Hello!

I have a Repeater form field containing a number of other fields. I want to allow deletion of Repeater entry (I will call such group of fields that are created when `addAction` is triggered) but only under certain condition = one of the fields has certain value. 

How can this be achieved / can this be achieved at all? 

I don't care how such goal is accomplished - the button for one particular entry can be hidden, or it can display, but when trying to perform an action for given entry the action should fail

What I have tried so far:

1. `deleteAction`, `before` and `abort`
```
Repeater::make('my_repeater')->schema(...)->deleteAction(fn (Action $action) => $action->before(function($arguments){
   $value = collect($this->data['my_repeater'])->filter(
      fn (array $_value, string $key) => Str::contains($arguments['item'], $key)
   )->flatten()->filter()->first();

   if ($value['my_field'] === 'my_value') ? abort(422, "Foo") : null;
})
```

This works, but `abort` is raised and I don't really want to throw exception to prevent an action from happening; without `abort` standard filament action will trigger and entry will be deleted (I don't want this); I'd be happy if there was some way to define that native delete action shouldn't trigger

2. `deleteAction`, `before` and `setExtraAttributes`
```
Repeater::make('my_repeater')->schema(...)->deleteAction(fn (Action $action) => $action->before(function(Component $component, Action $action){
   $value = collect($this->data['my_repeater'])->filter(
      fn (array $_value, string $key) => Str::contains($component->getId(), $key)
   )->flatten()->filter()->first();

   if ($value['my_field'] === 'my_value') ? $action->setExtraAttributes([
      'class' => 'hidden'
   ]) : null
})
```

This doesn't work, because at the point when extra attributes are being added to the button the action is already resolving

3. Trying to customise `deleteAction`

This doesn't work, because at this point I'm operating on Action definition for all Repeater entries that will be created

Thanks!

---

After hurdles this is solution I've arrived to:

```
->deleteAction(
    fn (Action $action) => $action->before(
        function(array $arguments) {
            $value = collect($this->data['my_repeater'])->filter(
                fn (array $_value, string $key) => Str::contains($arguments['item'], $key)
            )->flatten()->filter()->first();

            if (my_condition_check()) {
                Notification::make()
                    ->warning()
                    ->body('Cannot do foo')
                    ->send();

                throw new Cancel();
            }
    })
)
```

Hiding / disabling delete per Repeater entry would be nicer from UX, but I don't think that's achievable

**Comment by smiliyas** [comment]
Have you tried
```php
$action->halt();
```
To automatically close confirmation modal, add this after halt
```php
$action->cancel();
```

**Reply by eithed** [reply]
Thanks for the response! I'm not using confirmation modal and given that `cancel` does throw Cancel you can say that I did :)
